### PR TITLE
Bug 18538. fix "driver does not support quoting" when using the drive…

### PR DESCRIPTION
**Pull request by xpohoc69**
…r pdo_odbc

| Q             | A
| ------------- | ---
| Is bugfix?    | ✔️
| New feature?  | ❌
| Breaks BC?    | ❌
| Tests pass?   | ✔️
| Fixed issues  | #18538

### CODE REVIEW

**Comment by xpohoc69** [comment]
```
There was 1 failure:

1) yiiunit\framework\caching\MemCacheTest::testExpireAdd
Failed asserting that false matches expected 'expire_testa'.

/home/runner/work/yii2/yii2/tests/framework/caching/CacheTestCase.php:211
/home/runner/work/yii2/yii2/tests/framework/caching/MemCacheTest.php:55
/home/runner/work/yii2/yii2/vendor/phpunit/phpunit/phpunit:52
```

@samdark Hi! I think test failed not from my edits

**Comment by samdark** [comment]
Yes, nevermind. That one is a bit unstable.

**Comment by samdark** [comment]
Would you please add a line for CHANGELOG.md?

**Review by samdark — APPROVED** [review]

**Comment by samdark** [comment]
Related to https://github.com/yiisoft/yii/pull/4348

**Comment by xpohoc69** [comment]
@samdark I updated changelog

**Comment by samdark** [comment]
Thanks.

File: framework/db/Schema.php [hunk 1]
```diff
@@ -460,7 +460,7 @@ public function quoteValue($str)
             return $str;
         }
 
-        if (($value = $this->db->getSlavePdo()->quote($str)) !== false) {
+        if (mb_stripos($this->db->dsn, 'odbc:') === false && ($value = $this->db->getSlavePdo()->quote($str)) !== false) {
```
Comment (marcovtwout) [inline framework/db/Schema.php:463]: @xpohoc69 The first check could be made more robust by calling getDriverName() https://github.com/yiisoft/yii2/blob/master/framework/db/Connection.php#L975

Comment (xpohoc69) [reply framework/db/Schema.php:463]: @marcovtwout getDriverName() return mssql, not odbc

Comment (marcovtwout) [reply framework/db/Schema.php:463]: Ah, so we can only rely on the DSN to check if we are using ODBC?

Comment (xpohoc69) [reply framework/db/Schema.php:463]: Yes, I think so

**Pull request merged**